src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -0,0 +1,486 @@
+    _firstClosedToken{},
+    _secondClosedToken{}
+{
+    _root = Controls::Grid{};
+    _AddControlToRoot(_control);
+
+    // Set the background of the pane to match that of the theme's default grid
+    // background. This way, we'll match the small underline under the tabs, and
+    // the UI will be consistent on bot light and dark modes.
+    auto res = Application::Current().Resources();
+    winrt::Windows::Foundation::IInspectable key = winrt::box_value(L"BackgroundGridThemeStyle");
+    if (res.HasKey(key))
+    {
+        winrt::Windows::Foundation::IInspectable g = res.Lookup(key);
+        winrt::Windows::UI::Xaml::Style style = g.try_as<winrt::Windows::UI::Xaml::Style>();
+        _root.Style(style);
+    }
+}
+
+Pane::~Pane()
+{
+}
+
+// Method Description:
+// - Adds a given Terminal Control to our root Grid. Also registers an event
+//   handler to know when that control closed.
+// Arguments:
+// - control: The new TermControl to use as the content of our root Grid.
+void Pane::_AddControlToRoot(TermControl control)
+{
+    _root.Children().Append(control.GetControl());
+
+    _connectionClosedToken = control.ConnectionClosed([=]() {
+        if (control.CloseOnExit())
+        {
+            // Fire our Closed event to tell our parent that we should be removed.
+            _closedHandlers();
+        }
+    });
+}
+
+// Method Description:
+// - Get the root UIElement of this pane. There may be a single TermControl as a
+//   child, or an entire tree of grids and panes as children of this element.
+// Return Value:
+// - the Grid acting as the root of this pane.
+Controls::Grid Pane::GetRootElement()
+{
+    return _root;
+}
+
+// Method Description:
+// - Returns nullptr if no children of this pane were the last control to be
+//   focused, or the TermControl that _was_ the last control to be focused (if
+//   there was one).
+// - This control might not currently be focused, if the tab itself is not
+//   currently focused.
+// Return Value:
+// - nullptr if no children were marked `_lastFocused`, else the TermControl
+//   that was last focused.
+TermControl Pane::GetLastFocusedTerminalControl()
+{
+    if (_IsLeaf())
+    {
+        return _lastFocused ? _control : nullptr;
+    }
+    else
+    {
+        auto firstFocused = _firstChild->GetLastFocusedTerminalControl();
+        if (firstFocused != nullptr)
+        {
+            return firstFocused;
+        }
+        auto secondFocused = _secondChild->GetLastFocusedTerminalControl();
+        return secondFocused;
+    }
+}
+
+// Method Description:
+// - Returns nullopt if no children of this pane were the last control to be
+//   focused, or the GUID of the profile of the last control to be focused (if
+//   there was one).
+// Return Value:
+// - nullopt if no children of this pane were the last control to be
+//   focused, else the GUID of the profile of the last control to be focused
+std::optional<GUID> Pane::GetLastFocusedProfile() const noexcept

```

Comment:
can these two be simplified into `GetLastFocusedPane()` which does the descent logic, and then ask that one for its profile/control?

Reply:
You know, i was very tempted to do just that. I can dig it.